### PR TITLE
[Python Otel] Remove opentelemetry version pin

### PR DESCRIPTION
**Pull request by XuanWang-Amos**

### CODE REVIEW

File: requirements.bazel.txt [hunk 1]
```diff
@@ -26,8 +26,8 @@ pyasn1==0.5.0
 rsa==4.9
 greenlet==1.1.3.post0
 zope.interface==6.1
-opentelemetry-sdk==1.21.0
-opentelemetry-api==1.21.0
```
Comment (gnossen) [inline requirements.bazel.txt:30]: Let's keep the pin for Bazel unless we have a strong reason not to. We want the testing to be deterministic.

File: src/python/grpcio_tests/tests/observability/BUILD.bazel [hunk 2]
```diff
@@ -34,6 +37,7 @@ py_test(
         "//src/python/grpcio/grpc:grpcio",
         "//src/python/grpcio_observability/grpc_observability:pyobservability",
         "//src/python/grpcio_tests/tests/testing",
+        requirement("opentelemetry-sdk"),
```
Comment (gnossen) [inline src/python/grpcio_tests/tests/observability/BUILD.bazel:40]: Interesting. Why is this dependency necessary now?

Comment (XuanWang-Amos) [reply src/python/grpcio_tests/tests/observability/BUILD.bazel:40]: I removed sdk requirement from py_library `_opentelemetry_observability`, but sdk is still required in our testing.

Comment (gnossen) [reply src/python/grpcio_tests/tests/observability/BUILD.bazel:40]: Ah. This makes a lot of sense. Good change! :+1:

**Review by gnossen — APPROVED** [review]

**Pull request closed without merging**